lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1499,15 +1503,15 @@ OpFoldResult ShrPrimOp::fold(FoldAdaptor adaptor) {
 }
 
 LogicalResult ShrPrimOp::canonicalize(ShrPrimOp op, PatternRewriter &rewriter) {
-  auto inputWidth = op.getInput().getType().getWidthOrSentinel();
+  auto inputWidth = op.getInput().getType().get().getWidthOrSentinel();
```

Comment:
Is there literally a single place we don't unwrap the type (or wouldn't be equivalent if we did, since don't need to unwrap to use FIRRTLBaseType accessors, at least those not asking if it's an anonymous type)?

Don't have concrete suggestion how to act on that, since we don't also control enough of the code generation MLIR-side to emit our own `TypedValueOrAlias<T>` or something that makes this transparent, but does increasingly feel like these aren't actually types at all since we never actually want them or use them that way, but don't have a better way to track names in the way we want?
(so are inserting the names into type hierarchy as fake-types and being careful to step through them as the metadata indirection they are, to the real type...)